Black-Market/Black-Market/Network/NetworkService.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  NetworkService.swift
+//  Black-Market
+//
+//  Created by Junior Sancho on 8/2/23.
+//
+
+import Foundation
+import Combine
+
+class NetworkService {
+    func getData(from endpoint: NetworkEndpoint) -> AnyPublisher<Data, Error> {
+        guard let request = buildRequest(from: endpoint) else {
+            let error = NetworkError.network(description: "Couldn't create URLRequest")
+            return Fail(error: error)
+                .eraseToAnyPublisher()
+        }
+        
+        return loadData(with: request)
+            .eraseToAnyPublisher()
+    }
+    
+    private func buildRequest(from endpoint: NetworkEndpoint) -> URLRequest? {
+        guard var urlComponents = URLComponents(string: endpoint.absoluteURL) else {
+            return nil
+        }
+        if endpoint.httpMethod == .get {
+            urlComponents.queryItems = endpoint.params.compactMap({ param -> URLQueryItem in
+                return URLQueryItem(name: param.key, value: param.value as? String ?? "")
+            })
+        }
+        guard let url = urlComponents.url else {
+            return nil
+        }
+        var urlRequest = URLRequest(url: url,
+                                    cachePolicy: .reloadIgnoringLocalCacheData,
+                                    timeoutInterval: 30)
+        urlRequest.httpMethod = endpoint.httpMethod.rawValue
+        if endpoint.httpMethod == .post {
```

Comment:
shouldn't params be added to the body in PUT, PATCH and maybe DELETE(this is a controvesial one) as well?

Reply:
I will add this support in the next PR. I will analyze these cases.